### PR TITLE
Add option to print variable values and messages

### LINKED ISSUE #28: Add option to print variable values and messages

### PR DESCRIPTION
**Pull request by rotty3000**
fixes #28

### CODE REVIEW

**Comment by rotty3000** [comment]
As is, given the following plan:
```yaml
# Plan with variables and their output
variables:
  foo: bar
  dyn: "{{ 20 % 7 }}"
  fiz: 12
  bar:
  - 1
  - 2
  baz:
    name: George
    age: 45
requests:
- get:
    url: https://google.com
  output:
  - variables
```

this produces (notice that the template is not expanded):

image

**Comment by rotty3000** [comment]
needs tests

**Comment by rotty3000** [comment]
added test

**Comment by rotty3000** [comment]
As of now, given the following plan:
```yaml
# Plan with variables and their output
variables:
  foo: bar
  dyn: "{{ 20 % 7 }}"
  fiz: 12
  bar:
  - 1
  - 2
  baz:
    name:
      first: George
      last: Harrison
    age: 45
requests:
- get:
    url: https://google.com
  assert:
  - name: Check variable bar[1] == 2
    expression: bar[1] == 2
  - name: Check variable baz.name.first == 'George'
    expression: baz.name.first == 'George'
  output:
  - variables
```
the output is:

image

File: yaml_requests/utils/template.py [hunk 1]
```diff
@@ -30,37 +34,35 @@ def _contains_template(self, str_in):
...
-            except Exception:
-                pass
-
-        return rendered
-
-    def _resolve_dict(self, item):
+                self._json_decoder.scan_once(rendered, 0)
+            except json.JSONDecodeError:
+                # it's yaml
+                is_yaml = True
```
Comment (kangasta) [inline yaml_requests/utils/template.py]: This variable seems to be unused, could be replaced with `pass`.

File: yaml_requests/utils/template.py [hunk 2]
```diff
@@ -30,37 +34,35 @@ def _contains_template(self, str_in):
...
+            except StopIteration:
+                # it's not JSON
+                return rendered
+
+            return yaml.load(rendered, Loader=yaml.SafeLoader)
+        except Exception:
+            pass
```
Comment (kangasta) [inline yaml_requests/utils/template.py]: Should this also be `return rendered`? I guess the execution would end up here if both JSON and YAML load fail, but not sure when that would happen 🤔

File: yaml_requests/utils/template.py [hunk 3]
```diff
@@ -30,37 +34,35 @@ def _contains_template(self, str_in):
...
-        single_start_eq = str_in.count(self.variable_start_string) == 1
-        return start_eq and single_start_eq and end_eq
-
-    def _resolve_string(self, str_in):
+    def _resolve_string(self, str_in) -> any:
         if not self._contains_template(str_in):
             return str_in
 
-        if self._is_template(str_in):
-            str_in = str_in.replace('}}', ' | to_json }}')
```
Comment (kangasta) [inline yaml_requests/utils/template.py]: Adding ` | to_json` to the variable maintains the original data type of the variable. But, as you commented in the issue, there is a bug here on `main` as `| to_json` has higher precedence than some operators (e.g., `%`).

I'll create PR to demo this: #44

Comment (kangasta) [reply yaml_requests/utils/template.py]: There probably would be better way to maintain the data type than this `| to_json` hack, though.

**Comment by kangasta** [comment]
I rebased this on top of #44 and reverted the template handling change for now to maintain data-types of the variables when value is template. Would be good to get rid of my `|| to_json` hack at some point, though.

 I'll merge this and create a new release. Thank you for the implementation!

**Pull request merged**